src/data_provider/tests/sysInfoPackageLinuxParserRpm/sysInfoPackageLinuxParserRPM_test.cpp

```diff
@@ -0,0 +1,504 @@
+{
+    return m_libdb_mock->open(db, txnid, file, database, type, flags, mode);
+}
+int db_cursor(DB* db, DB_TXN* txnid, DBC** cursorp, u_int32_t flags)
+{
+    return m_libdb_mock->cursor(db, txnid, cursorp, flags);
+}
+int db_c_get(DBC* cursor, DBT* key, DBT* data, u_int32_t flags)
+{
+    return m_libdb_mock->c_get(cursor, key, data, flags);
+}
+int db_c_close(DBC* cursor)
+{
+    return m_libdb_mock->c_close(cursor);
+}
+int db_close(DB* db, u_int32_t flags)
+{
+    return m_libdb_mock->close(db, flags);
+}
+
+
+
+class CallbackMock
+{
+    public:
+        CallbackMock() = default;
+        ~CallbackMock() = default;
+        MOCK_METHOD(void, callbackMock, (nlohmann::json&), ());
+};
+
+
+
+TEST(SysInfoPackageLinuxParserRPM_test, rpmFromBerkleyDB)
+{
+    CallbackMock wrapper;
+
+    auto expectedPackage1 =
+        R"({"architecture":"amd64","description":"The Open Source Security Platform","format":"rpm","groups":"test","install_time":"5","name":"Wazuh","size":321,"vendor":"The Wazuh Team","version":"123:4.4-1"})"_json;
+
+    auto utils_mock { std::make_unique<UtilsMock>() };
+    auto libdb_mock { std::make_unique<LibDBMock>() };
+
+    m_utils_mock = utils_mock.get();
+    m_libdb_mock = libdb_mock.get();
+
+    DB db {};
+    DBC cursor {};
+
+    db.set_lorder = db_set_lorder;
+    db.open = db_open;
+    db.cursor = db_cursor;
+    db.close = db_close;
+    cursor.c_get = db_c_get;
+    cursor.c_close = db_c_close;
+
+    EXPECT_CALL(*utils_mock, existsRegular(_)).Times(1).WillOnce(Return(true));
+    EXPECT_CALL(*libdb_mock, db_create(_, _, _)).Times(1).WillOnce(DoAll(SetArgPointee<0>(&db), Return(0)));
+    EXPECT_CALL(*libdb_mock, set_lorder(_, _)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(*libdb_mock, open(_, _, _, _, _, _, _)).Times(1).WillOnce(Return(0));
+    EXPECT_CALL(*libdb_mock, cursor(_, _, _, _)).Times(1).WillOnce(DoAll(SetArgPointee<2>(&cursor), Return(0)));
+
+    // Emulate data stored in database
+
+    std::string name { "Wazuh" };
+    std::string version { "4.4" };
+    std::string release { "1" };
+    int epoch { 123 };
+    std::string summary { "The Open Source Security Platform" };
+    int itime { 5 };
+    int size { 321 };
+    std::string vendor { "The Wazuh Team" };
+    std::string group { "test" };
+    std::string source { "github" };
+    std::string arch { "amd64" };
+
+    const auto total_fields {11};
+    const auto total_fields_len
+    {
+        (name.length() + version.length() + release.length() + summary.length() + vendor.length() + group.length() + source.length() + arch.length() + 9) +
+        (sizeof(epoch) + sizeof(itime) + sizeof(size))
+    };
+
+    const auto total_len {FIRST_ENTRY_OFFSET + ENTRY_SIZE* total_fields + total_fields_len + 1};
+
+    DBT data, key;
+    memset(&key, 0, sizeof(key));
+    memset(&data, 0, sizeof(data));
+    char bytes[total_len];
+    memset(bytes, 0, sizeof(bytes));
```

Comment:
Hi @jnasselle 

I think is better to initialize the variable like this `char bytes[total_len] = { 0 }` instead of memset.

Best regards
Hanes